components/WorkoutsList/index.js

```diff
@@ -2,15 +2,61 @@ import Link from "next/link";
 import WorkoutForm from "../WorkoutForm";
 import WorkoutPreview from "../WorkoutPreview";
 import styled from "styled-components";
+import ModalDelete from "../ModalDelete";
+import { useState } from "react";
+
+export default function WorkoutsList({
+  onAddWorkout,
+  onDeleteWorkout,
+  workouts,
+  exercises,
+}) {
+  const [modalDelete, setModalDelete] = useState(false);
+  const [workoutIdToDelete, setWorkoutIdToDelete] = useState(0);
+  const [isWorkoutDeleted, setIsWorkoutDeleted] = useState(false);
```

Comment:
I think you don't need these useStates. Deleting a workout is done using the onDeleteWorkout function (in _app.js). This will of cause trigger a new render with the new `workouts`.